narwhal/primary/src/core.rs

```diff
@@ -580,13 +579,12 @@ impl Core {
             self.gc_round < header.round,
             DagError::TooOld(header.id.into(), header.round, self.gc_round)
         );
-        // TODO: enable below.
         // The header round is too high for this node, which is unlikely to acquire all
-        // parent certificates in time.
-        // ensure!(
-        //     self.highest_processed_round + MAX_HEADER_ROUND_CATCHUP_THRESHOLD > header.round,
-        //     DagError::TooNew(header.id.into(), header.round, self.gc_round)
-        // );
+        // parent certificates in time for voting.
+        ensure!(
+            self.highest_processed_round + MAX_HEADER_ROUND_CATCHUP_THRESHOLD > header.round,
+            DagError::TooNew(header.id.into(), header.round, self.gc_round)
+        );
```

Comment:
This is dangerous, we need to make sure that `header.author` resends the header if it didn't manage to get enough votes